### PR TITLE
add explicit --auth-mode login to az storage commands

### PR DESCRIPTION
**Pull request by nojnhuh**
**What type of PR is this?**
/kind flake

**What this PR does / why we need it**:

This PR adds the `--auth-mode login` flag to `az storage` commands instead of relying on the `AZURE_STORAGE_AUTH_MODE` environment variable, which [previous testing](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-azure/5061/pull-cluster-api-provider-azure-conformance-custom-builds/1821311932200652800) in this PR shows isn't propagated to many of these commands and would have caused the tests to fail if it was working without https://github.com/kubernetes/k8s.io/pull/7168.

I'm hoping this eliminates flakes like https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-azure/5074/pull-cluster-api-provider-azure-e2e-aks/1823470453667139584
> `ERROR: Server failed to authenticate the request. Please refer to the information in the www-authenticate header.`

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

- [X] cherry-pick candidate

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by nojnhuh** [comment]
warning: I'll be spamming this until it actually triggers a build of k8s instead of reusing things.

@jackfrancis Could this kind of change wreak too much havoc or could I add this to help repro?
```diff
diff --git a/hack/util.sh b/hack/util.sh
index 32c219129..8bb19e5c7 100755
--- a/hack/util.sh
+++ b/hack/util.sh
@@ -30,15 +30,7 @@ capz::util::get_latest_ci_version() {
 }
 
 capz::util::should_build_kubernetes() {
-    if [[ -n "${TEST_K8S:-}" ]]; then
-        echo "true" && return
-    fi
-    # JOB_TYPE, REPO_OWNER, and REPO_NAME are environment variables set by a prow job -
-    # https://github.com/kubernetes/test-infra/blob/master/prow/jobs.md#job-environment-variables
-    if [[ "${JOB_TYPE:-}" == "presubmit" ]] && [[ "${REPO_OWNER:-}/${REPO_NAME:-}" == "kubernetes/kubernetes" ]]; then
-        echo "true" && return
-    fi
-    echo "false"
+    echo "true"
 }
 
 capz::util::should_build_ccm() {
```

**Comment by nojnhuh** [comment]
I think this should work once https://github.com/kubernetes/k8s.io/pull/7168 merges and the Terraform is reapplied.

**Comment by nojnhuh** [comment]
This is ready for review now. I can see the tests are at least getting past the setup phase in the tests I've checked which is all that this PR affects.

/retitle add explicit --auth-mode login to az storage commands
/cherry-pick release-1.16

File: scripts/ci-build-azure-ccm.sh [hunk 1]
```diff
@@ -66,17 +66,17 @@ main() {
...
         make -C "${AZURE_CLOUD_PROVIDER_ROOT}" bin/azure-acr-credential-provider bin/azure-acr-credential-provider.exe
 
-        if [[ "$(az storage container exists --name "${AZURE_BLOB_CONTAINER_NAME}" --query exists --output tsv)" == "false" ]]; then
+        if [[ "$(az storage container exists --name "${AZURE_BLOB_CONTAINER_NAME}" --query exists --output tsv --auth-mode login)" == "false" ]]; then
             echo "Creating ${AZURE_BLOB_CONTAINER_NAME} storage container"
-            az storage container create --name "${AZURE_BLOB_CONTAINER_NAME}" > /dev/null
+            az storage container create --name "${AZURE_BLOB_CONTAINER_NAME}" --auth-mode login > /dev/null
             # if the storage account has public access disabled at the account level this will return 404
-            az storage container set-permission --name "${AZURE_BLOB_CONTAINER_NAME}" --public-access container > /dev/null
+            az storage container set-permission --name "${AZURE_BLOB_CONTAINER_NAME}" --public-access container --auth-mode login > /dev/null
```
Comment (jsturtevant) [inline scripts/ci-build-azure-ccm.sh:73]: this one won't work 😢 

```
az storage container set-permission --auth-mode login --name test --public-access container
az storage container set-permission: 'login' is not a valid value for '--auth-mode'. Allowed values: key
```

will need to set an env variable for this one

**Review by jsturtevant — CHANGES_REQUESTED** [review]
lgtm, minus one small bug in azure cli

File: scripts/ci-build-azure-ccm.sh (hunk 1 above)
Comment (jsturtevant) [reply scripts/ci-build-azure-ccm.sh:73]: `AZURE_STORAGE_AUTH_MODE` will work for this command

Comment (nojnhuh) [reply scripts/ci-build-azure-ccm.sh]: Thanks! Fixed.

**Comment by jsturtevant** [comment]
tested this locally and it worked!

**Comment by nojnhuh** [comment]
This keeps flip flopping between passing in the VMSS test and the HA test but can't pass both for some reason.

/retest

**Review by mboersma — APPROVED** [review]
/lgtm
/approve

**Pull request merged**